api/src/main/java/org/hawkular/apm/api/services/Criteria.java

```diff
@@ -48,7 +50,6 @@
     private String hostName;
     private long upperBound;
     private long lowerBound;
-    private String principal;
```

Comment:
Did consider that, but it has actually been removed as a filter property from the UI - primarily I think because especially when using keycloak, the principal is a UUID rather than human meaningful value.

It can still be filtered as a property though.